File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here

new_array = []

arr = str.split('')

arr.each do |char| 
  new_array.unshift(char)
end

new_array.join

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')
  
  puts

  puts "Expecting: 'leahciM'"
  puts "=>", reverse_string('Michael')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  
  const new_array = []

  const array = str.split('')

  array.forEach(char => {
    new_array.unshift(char)
  });

  const new_word = new_array.join('')

  return new_word

}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
  
  console.log("");

  console.log("Expecting: 'leahciM'");
  console.log("=>", reverseString("Michael"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution
